src/hooks/useConfirmModal.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  useSetConfirmModalState,
+  ConfirmModalStateType,
+} from '@/atoms/confirmModalState';
+
+const useConfirmModal = (): readonly [
+  open: (next?: Partial<ConfirmModalStateType>) => void,
+  close: (next?: Partial<ConfirmModalStateType>) => void,
+  setConfirmModal: (next?: Partial<ConfirmModalStateType>) => void,
+] => {
+  const setConfirmModalState = useSetConfirmModalState();
+
+  const setConfirmModal = (next?: Partial<ConfirmModalStateType>) =>
+    setConfirmModalState((prev) => ({ ...prev, ...next }));
+
+  const open = (next?: Partial<ConfirmModalStateType>) =>
+    setConfirmModalState((prev) => ({ ...prev, ...next, isOpen: true }));
+
+  const close = (next?: Partial<ConfirmModalStateType>) =>
+    setConfirmModalState((prev) => ({ ...prev, next, isOpen: false }));
```

Comment:
open, close도 next 라는 인자를 받아야 할 경우가 있을까요??
없다면 지워주셔도 될 것 같습니다!

```suggestion
  const open = () =>
    setConfirmModalState((prev) => ({ ...prev, isOpen: true }));

  const close = () =>
    setConfirmModalState((prev) => ({ ...prev, isOpen: false }));
```

Reply:
상태가 바뀔 때마다 해당 상태를 사용하고 있는 컴포넌트는 리렌더링 연산이 발생하기 때문에 setConfirmModal이라는 함수를 호출해 message를 설정하고 open 함수로 modal을 보여주는 것보다 open이라는 함수에 인자를 받아 상태를 한번에 설정하는 것이 더 효율적이라고 생각했습니다.

그래서 open, close라는 함수에 인자를 추가한 것이고요. 추가적으로 이야기하자면 위에 제시한 예시코드에서부터 open, close에 인자를 넘겨서 사용하고 있습니당.

```ts
  const [open] = useConfirmModal();

  const modalCancelHandler = () => {
    open({ message: 'CANCEL', cancelHandler: null });
  };

  const openModal = () => {
    open({ message: 'HELLO', cancelHandler: modalCancelHandler });
  };
```

Reply:
아하~ 이런 의미가 있었군요 이해했습니다 수고하셨습니다!!